contrib/win-installer/test-installer.ps1

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env pwsh
+
+# The Param statement must be the first statement, except for comments and any #Require statements.
+param (
+    [Parameter(Mandatory)]
+    [ValidateSet("install", "uninstall", "all")]
+    [string]$operation,
+    [Parameter(Mandatory)]
+    [ValidateScript({Test-Path $_ -PathType Leaf})]
+    [string]$setupExePath,
+    [ValidateSet("wsl", "hyperv")]
+    [string]$provider="wsl",
+    [switch]$installWSL=$false,
+    [switch]$installHyperV=$false,
+    [switch]$skipWinVersionCheck=$false
+)
+
+$ConfFilePath = "$env:ProgramData\containers\containers.conf.d\99-podman-machine-provider.conf"
+$WindowsPathsToTest = @("C:\Program Files\RedHat\Podman\podman.exe",
+                        "C:\Program Files\RedHat\Podman\win-sshproxy.exe",
+                        "$ConfFilePath",
+                        "HKLM:\SOFTWARE\Red Hat\Podman")
+
+function Test-Installation {
+    $wslCheckboxVar = $($installWSL ? "1" : "0")
+    $hypervCheckboxVar = $($installHyperV ? "1" : "0")
+    $allowOldWinVar = $($skipWinVersionCheck ? "1" : "0")
```

Comment:
@l0rd these 3 lines seem to be failing tests.

Reply:
@lsm5 thank you. I suspect that's related to the version of PowerShell that's used by CI (< v7.0) that doesn't support the [ternary operator syntax](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_if?view=powershell-7.4#using-the-ternary-operator-syntax) I am going to put this PR back in draft mode until I fix it.

Reply:
are you planning to update the cirrus windows image with latest powershell? Looks like github actions windows env does have [required min-version of powershell](https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md#powershell-tools)

Reply:
This is an option, yes. But no, I think it's simpler to update the PowerShell script introduced with this PR to support old and new versions of PWSH. So that it has a wider compatibility of windows dev environments.